arrow-core-data/src/main/kotlin/arrow/core/MapK.kt

```diff
@@ -65,17 +65,22 @@ data class MapK<K, out A>(private val map: Map<K, A>) : MapKOf<K, A>, Map<K, A>
 
 fun <K, A> Map<K, A>.k(): MapK<K, A> = MapK(this)
 
+@Deprecated("Deprecated, use nullable instead", ReplaceWith("Tuple2<k, A>?.k()"))
 fun <K, A> Option<Tuple2<K, A>>.k(): MapK<K, A> =
   when (this) {
     is Some -> mapOf(this.t).k()
     is None -> emptyMap<K, A>().k()
   }
 
+fun <K, A> Tuple2<K, A>?.k(): MapK<K, A> =
+  this?.let { mapOf(this).k() } ?: emptyMap<K, A>().k()
+
```

Comment:
I believe this is not used, is it?

```suggestion
```

Reply:
It's a replacement for `Option<Tuple2<K, A>>.k()` above.

Reply:
I see, but I doubt this function should be exposed to the public (unless @nomisRev thinks otherwise) because it makes no sense name-wise and it should be internal. Now being internal, if we don't use it, then it can be completely removed I believe

Reply:
ah you are right, I will update it to have a better name and probably to be private.